### ISSUE TITLE
Category filters don't filter categories on Western Shield - Camera Watch project

### ISSUE THREAD

**Issue by yshish** [zooniverse/Panoptes-Front-End#4993]
## Expected behavior
When someone clicks on a category filter (Looks like / Pattern / Colour) the list of categories should filter out all animals that look differently. 

## Current behavior
Nothing happens when I click on any of the available filters. The category list stays the same, still showing 24 of 24 categories no matter what I choose. (I've just tested and the filters are working perfectly on Amazon Cam for me, so I guess it is just Western Shield situation.)

## Steps to replicate
Go to the classification and click on any of the filters while watching the list of categories. Check the number at the bottom 'Showing x from 24 categories.

## Additional information
No idea how long this happens as I have only tested them today to write down how to use them.

- **Operating system:** Ubuntu
- **Browser:** Chromium 69

**Comment by eatyourgreens** [comment]
This seems to be a problem with the modal form component that's used for the filter popup.

This is the error that's thrown when you try to select a filter on that project.
```
Uncaught Error: Unable to find node on an unmounted component.
    at invariant (react-dom.development.js:55)
    at findCurrentFiberUsingSlowPath (react-dom.development.js:4256)
    at findCurrentHostFiber (react-dom.development.js:4266)
    at findHostInstance (react-dom.development.js:17676)
    at Object.findDOMNode (react-dom.development.js:18145)
    at StickyModalForm.reposition (sticky.js:64)
```

It seems to be trying to find the component's DOM node before it has mounted. https://github.com/zooniverse/modal-form/blob/1e8276238f71576255f979519bf2d0c87eb19e4e/sticky.js#L61-L65

**Comment by eatyourgreens** [comment]
Other survey projects don't throw the same error, so this might be caused by a problem in the specific setup for Western Shield, which is then causing an error in the filter popup.

**Comment by eatyourgreens** [comment]
I think I've tracked this down to being caused by subjects with multiple images, when the browser is still waiting to load some frames. If you play through the subject slideshow once on Western Shield, so that all three images are loaded in the browser, then the filters work as expected.

Based on that, I think it's caused by this onLoad handler not being properly removed when the component unmounts:
https://github.com/zooniverse/modal-form/blob/1e8276238f71576255f979519bf2d0c87eb19e4e/sticky.js#L49-L50

ie. the filter form mounts, adds load handlers to all the images on the page, then unmounts when a filter is selected. Later, an image on the page finishes loading and calls `reposition` on the unmounted filter form, which throws an error.

**Comment by eatyourgreens** [comment]
React error boundaries can't catch [errors in event handlers](https://reactjs.org/docs/error-boundaries.html#how-about-event-handlers), so the `unable to find node` error would need to be fixed in [zooniverse/modal-form.](/zooniverse/modal-form)

**Comment by yshish** [comment]
> I think I've tracked this down to being caused by subjects with multiple images, when the browser is still waiting to load some frames. If you play through the subject slideshow once on Western Shield, so that all three images are loaded in the browser, then the filters work as expected.
> 
Thanks, Jim. Although I don't understand some of your replies in this thread, I am pretty sure the cause was elsewhere than in waiting for all images to get loaded. Because when I tested that, all images have been loaded for quite some time although the filters didn't work. I tested that multiple times on multiple devices with the same results I described.

Early this morning I found a reply from Birgus2 (the project owner) where she apologized for changing something in the rodent-like filter. As I understand that, it is what likely caused all the troubles: https://www.zooniverse.org/projects/birgus2/western-shield-camera-watch/talk/372/761674?comment=1264407 

I'm sorry for not letting you know about that earlier, I wasn't at the computer until now to reach you on GitHub. So hopefully you didn't spent too much time on that.

The filters are working for me now as expected. Not sure whether Birgus2 fixed that or you did. Anyway, I am very happy it is fixed, and appreciated all your efforts. Thanks! :)
